packages/config/src/context.js

```diff
@@ -87,9 +87,18 @@ const isEdgeFunctionsConfig = function (key, value) {
 // Takes into account that `context.{context}.build.*` is the same as
 // `context.{context}.*`
 export const ensureConfigPriority = function ({ build = {}, ...config }, context, branch) {
-  return {
+  const base = {
     ...config,
     build,
+  }
+
+  // remove the redirects to not have context specific redirects.
+  // The redirects should be only on the root level.
+  // eslint-disable-next-line fp/no-delete, no-param-reassign
+  delete config.redirects
```

Comment:
Might be nicer to use the destructuring to discard the `redirects` property rather than using `delete`?

```js
const { redirects: _, ...otherConfig } = config
```

(And then enable [`ignoreRestSiblings`](https://eslint.org/docs/latest/rules/no-unused-vars#ignorerestsiblings) so that ESLint is happy)

Reply:
IMHO this makes it harder to understand whats going on there (destructuring and not using) as the delete is very explicit.
Furthermore you instantiate and declare a variable that is completely unused.

But I can do it for you

Reply:
That's fair! In that case, we might want to revisit these ESLint rules in the future.

Reply:
To be honest I really would love to revisit this!

In my opinion Lint rules should help to avoid bugs and NOT enforce a certain code style. Code style is for me something that should be done by prettier otherwise it's just bikeshedding.

Reply:
Agreed!